quill-engine/src/main/scala/io/getquill/sql/idiom/VerifySqlQuery.scala

```diff
@@ -128,8 +128,8 @@ object VerifySqlQuery {
         case op: OptionTableForall if op.quat.isInstanceOf[Quat.Product] => throw new IllegalArgumentException("Cannot use Option.tableForAll on a table or embedded case class")
         case op: OptionTableExists if op.quat.isInstanceOf[Quat.Product] => throw new IllegalArgumentException("Cannot use Option.tableExists on a table or embedded case class")
 
-        case cond: If if cond.`then`.isInstanceOf[Quat.Product] => throw throw new IllegalArgumentException("Cannot use table or embedded case class as a result of a condition")
-        case cond: If if cond.`else`.isInstanceOf[Quat.Product] => throw throw new IllegalArgumentException("Cannot use table or embedded case class as a result of a condition")
+        case cond: If if cond.`then`.isInstanceOf[Quat.Product] => throw new IllegalArgumentException("Cannot use table or embedded case class as a result of a condition")
+        case cond: If if cond.`else`.isInstanceOf[Quat.Product] => throw new IllegalArgumentException("Cannot use table or embedded case class as a result of a condition")
```

Comment:
Nothing exploded when I made this change. But flagging because neither my scala aware peer nor I understood what the current code was trying to do.